src/workerd/api/worker-rpc.c++

```diff
@@ -147,12 +186,107 @@ private:
   kj::ForkedPromise<void> promise;
 };
 
+// Given a value, check if it has a dispose method and, if so, invoke it.
+void tryCallDisposeMethod(jsg::Lock& js, jsg::JsValue value) {
+  js.withinHandleScope([&]() {
```

Comment:
non-blocking: Hopefully v8 will give us an api to call this directly on a v8::Object